From patchwork Thu Mar 28 17:10:00 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Paolo Bonzini <pbonzini@redhat.com>
X-Patchwork-Id: 10875443
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 12B2613B5
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 28 Mar 2019 17:10:24 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E707828BE7
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 28 Mar 2019 17:10:23 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E51B028CDF; Thu, 28 Mar 2019 17:10:23 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7040728BE7
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 28 Mar 2019 17:10:23 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726811AbfC1RKF (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 28 Mar 2019 13:10:05 -0400
Received: from mail-wr1-f68.google.com ([209.85.221.68]:36180 "EHLO
        mail-wr1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725816AbfC1RKF (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 28 Mar 2019 13:10:05 -0400
Received: by mail-wr1-f68.google.com with SMTP id y13so6166827wrd.3;
        Thu, 28 Mar 2019 10:10:03 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=sender:from:to:subject:date:message-id;
        bh=nPCFzUJnrLwfTHXgaehhT7OlM6k66qhIj2xZuLJSMts=;
        b=oYS2q/11oi1zXmxooJ/F7NrroxcvAKk5OWoSFNApar9dlhdBiowZCyqedGhYll5+BL
         qtP3+7bKCqMhq+0l3KZ7UmBfcuQmcvoSlVbSTYqNpw3mvKh1WT7keCyqo7RMddvVQXk2
         /UyTaiHbE1s+T/Szhwyoxja8DVvqzJmCwbmrAe7rbRbjil142r5uNUYjEmX1VDlY6s+s
         PnhmYbpW+P35uqHefbR64GdN6zWeILnV+vw+fnq/7YTW3VzeSeLJdNH1v2uacIrFaxdM
         6N7oh1eCO64Ncx6uRUh99tc1qPWsYukUjZDuXMFRMziN8LpUgfNKRwdt/Rq1faWszqRS
         pFZQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:from:to:subject:date:message-id;
        bh=nPCFzUJnrLwfTHXgaehhT7OlM6k66qhIj2xZuLJSMts=;
        b=VooN9CbnKjh3ANOWSejAVnoHYxxF3HP0L+RfJgWpSfn1Ay/HtBVXak7g6PmlATemnV
         IE/O/Fjj/cTlz4PxDLGVAXJasiy7j+kLsXgulQ2OmDzVyCSFmF48vZE6ZS+N9wcRrEKR
         N7tqWb9Q3CKeF0NkkJ2ncKWcqUNyPtrHb/1RSLyjpY5lMmkl2mr2DGt/3nzqvd1TbQTn
         eDQ4pW9gw8KfJGvfdHRNpHynIfMRTZxZ8Psd+THforzQlkjN4aqVFFaub82NI2VohE7X
         7y7kY0n25Le19oE/RBxz5kyC4GUXEmMDMOegKOqJtIapdKoEe+sOdxRkxsiPLVfpyUxJ
         1Pog==
X-Gm-Message-State: APjAAAVGgcDpibA5wp46yP8CUQpcIHTAOAy/visvT4FKZyl0YvahmkGr
        8fLReqOWcz4JUI8QGix3CkvVejjw
X-Google-Smtp-Source: 
 APXvYqwyjkrRNK9VDO9y9amfAdPu744a+CjdPpkV2a5xwhli1XPIIStcDFBOP0mYukGFsdF3EMaalg==
X-Received: by 2002:adf:eb02:: with SMTP id s2mr9189981wrn.29.1553793002606;
        Thu, 28 Mar 2019 10:10:02 -0700 (PDT)
Received: from 640k.localdomain ([93.56.166.5])
        by smtp.gmail.com with ESMTPSA id
 a82sm4803048wmf.11.2019.03.28.10.10.01
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Thu, 28 Mar 2019 10:10:01 -0700 (PDT)
From: Paolo Bonzini <pbonzini@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Subject: [PATCH] kvm: move KVM_CAP_NR_MEMSLOTS to common code
Date: Thu, 28 Mar 2019 18:10:00 +0100
Message-Id: <1553793000-53968-1-git-send-email-pbonzini@redhat.com>
X-Mailer: git-send-email 1.8.3.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

All architectures except MIPS were defining it in the same way,
and memory slots are handled entirely by common code so there
is no point in keeping the definition per-architecture.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 Documentation/virtual/kvm/api.txt | 5 ++---
 arch/powerpc/kvm/powerpc.c        | 3 ---
 arch/s390/kvm/kvm-s390.c          | 3 ---
 arch/x86/kvm/x86.c                | 3 ---
 virt/kvm/arm/arm.c                | 3 ---
 virt/kvm/kvm_main.c               | 2 ++
 6 files changed, 4 insertions(+), 15 deletions(-)

diff --git a/Documentation/virtual/kvm/api.txt b/Documentation/virtual/kvm/api.txt
index 67068c47c591..b62ad0d94234 100644
--- a/Documentation/virtual/kvm/api.txt
+++ b/Documentation/virtual/kvm/api.txt
@@ -1117,9 +1117,8 @@ struct kvm_userspace_memory_region {
 This ioctl allows the user to create, modify or delete a guest physical
 memory slot.  Bits 0-15 of "slot" specify the slot id and this value
 should be less than the maximum number of user memory slots supported per
-VM.  The maximum allowed slots can be queried using KVM_CAP_NR_MEMSLOTS,
-if this capability is supported by the architecture.  Slots may not
-overlap in guest physical address space.
+VM.  The maximum allowed slots can be queried using KVM_CAP_NR_MEMSLOTS.
+Slots may not overlap in guest physical address space.
 
 If KVM_CAP_MULTI_ADDRESS_SPACE is available, bits 16-31 of "slot"
 specifies the address space which is being modified.  They must be
diff --git a/arch/powerpc/kvm/powerpc.c b/arch/powerpc/kvm/powerpc.c
index 8885377ec3e0..92910b7c5bcc 100644
--- a/arch/powerpc/kvm/powerpc.c
+++ b/arch/powerpc/kvm/powerpc.c
@@ -644,9 +644,6 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 		else
 			r = num_online_cpus();
 		break;
-	case KVM_CAP_NR_MEMSLOTS:
-		r = KVM_USER_MEM_SLOTS;
-		break;
 	case KVM_CAP_MAX_VCPUS:
 		r = KVM_MAX_VCPUS;
 		break;
diff --git a/arch/s390/kvm/kvm-s390.c b/arch/s390/kvm/kvm-s390.c
index 4638303ba6a8..28f35d2b06cb 100644
--- a/arch/s390/kvm/kvm-s390.c
+++ b/arch/s390/kvm/kvm-s390.c
@@ -513,9 +513,6 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 		else if (sclp.has_esca && sclp.has_64bscao)
 			r = KVM_S390_ESCA_CPU_SLOTS;
 		break;
-	case KVM_CAP_NR_MEMSLOTS:
-		r = KVM_USER_MEM_SLOTS;
-		break;
 	case KVM_CAP_S390_COW:
 		r = MACHINE_HAS_ESOP;
 		break;
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 099b851dabaf..9b64d3359c93 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -3073,9 +3073,6 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 	case KVM_CAP_MAX_VCPUS:
 		r = KVM_MAX_VCPUS;
 		break;
-	case KVM_CAP_NR_MEMSLOTS:
-		r = KVM_USER_MEM_SLOTS;
-		break;
 	case KVM_CAP_PV_MMU:	/* obsolete */
 		r = 0;
 		break;
diff --git a/virt/kvm/arm/arm.c b/virt/kvm/arm/arm.c
index 99c37384ba7b..be4ec5f3ba5f 100644
--- a/virt/kvm/arm/arm.c
+++ b/virt/kvm/arm/arm.c
@@ -224,9 +224,6 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 	case KVM_CAP_MAX_VCPUS:
 		r = KVM_MAX_VCPUS;
 		break;
-	case KVM_CAP_NR_MEMSLOTS:
-		r = KVM_USER_MEM_SLOTS;
-		break;
 	case KVM_CAP_MSI_DEVID:
 		if (!kvm)
 			r = -EINVAL;
diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 55fe8e20d8fd..31c28e0b067c 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -3061,6 +3061,8 @@ static long kvm_vm_ioctl_check_extension_generic(struct kvm *kvm, long arg)
 #endif
 	case KVM_CAP_MAX_VCPU_ID:
 		return KVM_MAX_VCPU_ID;
+	case KVM_CAP_NR_MEMSLOTS:
+		return KVM_USER_MEM_SLOTS;
 	default:
 		break;
 	}
